BTrees/_base.py

```diff
@@ -815,9 +815,7 @@ def __delitem__(self, key):
         self._del(self._to_key(key))
 
     def clear(self):
-        if self._data:
-            # In the case of __init__, this was already set by __new__
-            self._data = []
+        self._data = []
```

Comment:
I see, thanks. So it is not clear-specific but just a general things about variables in slots. My approach, once again, would be to fix those cases one-by-one with, as you suggested, bumping persistent version for the slots ghostification patch.

It would be a pity not to be able to rely on ghostification semantics to completely release objects.